libweb3jsonrpc/Test.cpp

```diff
@@ -23,16 +23,52 @@
 #include "Test.h"
 #include <jsonrpccpp/common/errors.h>
 #include <jsonrpccpp/common/exception.h>
-#include <libethereum/ClientTest.h>
+#include <libdevcore/CommonJS.h>
 #include <libethereum/ChainParams.h>
+#include <libethereum/ClientTest.h>
 
 using namespace std;
 using namespace dev;
+using namespace dev::eth;
 using namespace dev::rpc;
 using namespace jsonrpc;
 
 Test::Test(eth::Client& _eth): m_eth(_eth) {}
 
+namespace
+{
+string logEntriesToLogHash(eth::LogEntries const& _logs)
+{
+    RLPStream s;
+    s.appendList(_logs.size());
+    for (eth::LogEntry const& l : _logs)
+        l.streamRLP(s);
+    return toHexPrefixed(sha3(s.out()));
+}
+}
+
+string Test::test_getLogHash(string const& _txHash)
+{
+    h256 txHash;
+    try
+    {
+        txHash = h256(_txHash);
+    }
+    catch (...)
+    {
+        BOOST_THROW_EXCEPTION(JsonRpcException(Errors::ERROR_RPC_INVALID_PARAMS));
+    }
+
+    if (m_eth.blockChain().isKnownTransaction(txHash))
+    {
+        LocalisedTransaction t = m_eth.localisedTransaction(txHash);
```

Comment:
This could throw, too (that would mean something is not right in DB), so I would wrap all of this `if` in `try..catch`, too and rethrow as `JsonRpcException(Errors::ERROR_RPC_INTERNAL_ERROR)`

Reply:
Yeap, I thought of that